core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1973,6 +1971,7 @@ public static ExpressionAnalysis analyzeExpression(
         analyzer.analyze(expression, scope);
 
         updateAnalysis(analysis, analyzer, session, accessControl);
+        analysis.addReferencedFields(analyzer.getReferencedFields(), expression);
```

Comment:
What's the use of mapping referenced fields by the top-level analyzed expression? Couldn't we map them by the sub-expression which is the actual column reference? If so, it would be sufficient to do:
`originColumnDetails.put(NodeRef.of(node), ImmutableList.of(field.getOriginColumnDetails))` in the method `handleResolvedField()`.

Reply:
OK, got it.